### ISSUE TITLE
SC2029 does not catch single quotes within double quotes

### ISSUE THREAD

**Issue by fenollp** [koalaman/shellcheck#573]
```
In ./kerl line 582:
    ssh $KERL_DEPLOY_SSH_OPTIONS "$host" "cd '$remotepath' && sed -i -e 's#$path#$(pwd)#g' activate"
        ^-- SC2086: Double quote to prevent globbing and word splitting.
                                              ^-- SC2029: Note that, unescaped, this expands on the client side.
```

I don't believe `$remotepath` will actually get expanded on the client side, given the single quotes.

**Comment by koalaman** [comment]
ShellCheck is right. Single quotes are literal in double quotes, and therefore have no quoting behavior.

All the variable expansions in your expression will happen on the client, and the server will never see nor expand `$remotepath`, `$path` or `$(pwd)`. I don't know where these are set, but the latter one definitely looks like it should happen on the server side, so it should be `\$(pwd)`

**Comment by fenollp** [comment]
Oh my god you're right

**Comment by nchelluri** [comment]
I'm just browsing the repo trying to understand one of the rules (the incorrect quoting example from the readme labeled `# Literal quotes in variables`) and ended up reading this page, I just wanted to say @fenollp even though we've never chatted somehow I can imagine the tone of shock and surprise in your reply ("Oh my god...") and it gave me a smile, I've had plenty of those moments myself :)

**Comment by bittorf** [comment]
@nchelluri basically because it is no valid option string.
check both commands below and you should understand the issue:

have a simple file test.sh with:
```
#!/bin/sh
echo "$1"
```
so the parser simply gets confused:
```
user@box:~$ ./test.sh '--verbose="true"'
--verbose="true"
user@box:~$ ./test.sh --verbose="true"
--verbose=true
```

**Comment by nchelluri** [comment]
Thanks @bittorf .
